src/databricks/labs/ucx/hive_metastore/catalog_schema.py

```diff
@@ -23,6 +32,30 @@ def create_all_catalogs_schemas(self, prompts: Prompts):
         for candidate_catalog, schemas in candidate_schemas.items():
             for candidate_schema in schemas:
                 self._create_schema(candidate_catalog, candidate_schema)
+        self._update_principal_acl()
+
+
+    def _get_catalog_schema_grants(self):
+        catalog_grants: set[Grant] = set()
+        src_trg_schema_mapping = self._get_src_database_target_mapping()
+        grants = self._principal_grants.get_interactive_cluster_grants()
+        database_grants = [grant for grant in grants if grant.table is None or grant.view is None]
```

Comment:
There's type_and_key() method on a grant that you should be using

Reply:
I taking the output from self._principal_grants.get_interactive_cluster_grants() and filtering for only database level grants